### ISSUE TITLE
IndexError while using sparse_nmf initialization method on demo_dendritic.ipynb

### ISSUE THREAD

**Issue by KunyunWang** [flatironinstitute/CaImAn#1076]
Please fill in the following for any issues

Your setup:
 1. Operating System (Linux, MacOS, Windows): MacOS
 2. Hardware type (x86, ARM..) and RAM: M1, 16G
 3. Python Version (e.g. 3.9): 3.10.10
 4. Caiman version (e.g. 1.9.12): 1.9.15
 5. Which demo exhibits the problem (if applicable): demo_dendritic.ipynb
 6. How you installed Caiman (pure conda, conda + compile, colab, ..): pure conda
 7. Details: When I was doing refitting using sparse_nmf initialization method, there is an index error: IndexError: index 60 is out of bounds for axis 0 with size 60 (in this demo, we extract 60 components). I think it might have something to do with the extracted components during fitting since I found 2 null components after the first fitting.

**Comment by EricThomson** [comment]
Thanks for bringing this to our attention we are looking into it and let you know when we have figured out what is going on.

**Comment by pgunn** [comment]
Confirmed on Linux with a fresh install, replacing the value assigned to `method_init` in the 4th cell with `sparse_nmf`.

I think it starts to have problems earlier, perhaps in the call to cnm.fit(), where we start to see warnings about encountering invalid values.

**Comment by EricThomson** [comment]
I've done some digging, and I would be wary of using `sparse_nmf` right now. There were some changes made to scikit-learn's NMF algorithm that this initialization depends on (see discussion here: https://github.com/scikit-learn/scikit-learn/issues/4811). 

`graph_nmf seems better behaved, and while it depends on scikit-learn's NMF function, it was was written after the changes above were implemented.

I'll post more about this after I've researched it more and had time to figure out exactly what's going on, but we would also welcome some help with it.
